### PR TITLE
[Orchestrator] Run orchestrator_pod check by default

### PR DESCRIPTION
**Pull request by JLineaweaver**
### What does this PR do?

Follow up to https://github.com/DataDog/datadog-agent/pull/18740 that creates the configuration to run the check by default

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: cmd/agent/dist/conf.d/orchestrator_pod.d/conf.yaml.default
Comment (sgnn7) [inline cmd/agent/dist/conf.d/orchestrator_pod.d/conf.yaml.default:3]: Could we keep this format consistent with the [other default files](https://github.com/DataDog/datadog-agent/pull/19478/files)?

**Review by vickenty — CHANGES_REQUESTED** [review]
We need to fix [the errors if the check is not enabled](https://github.com/DataDog/datadog-agent/pull/18740#discussion_r1317163422) before merging this PR.

File: cmd/agent/dist/conf.d/orchestrator_pod.d/conf.yaml.default
Comment (JLineaweaver) [reply cmd/agent/dist/conf.d/orchestrator_pod.d/conf.yaml.default]: [66510cc](https://github.com/DataDog/datadog-agent/pull/19478/commits/66510cc8c48c5f1d234ff105baab88101a775713)

File: releasenotes/notes/AddPodCheck-355c270d2179d511.yaml
Comment (buraizu) [inline releasenotes/notes/AddPodCheck-355c270d2179d511.yaml:11]:
```suggestion
    Introducing the new orchestrator pod check that now resides in corechecks. In the next release, this new check will replace the current pod check in the Process Agent. You can start using this new check now by manually setting the environment variable `DD_ORCHESTRATOR_EXPLORER_RUN_ON_NODE_AGENT` to `true`.
```

**Review by buraizu — APPROVED** [review]
Thanks, approving with some minor edits requested on the release note

File: pkg/process/checks/pod.go
Comment (mfpierre) [inline pkg/process/checks/pod.go:62]: The concept of core checks feels like an implementation detail that users are probably not aware of

```suggestion
	log.Warn("This Process Agent check will be deprecated in 7.51.0 and moved to the Node Agent")
```

File: releasenotes/notes/AddPodCheck-355c270d2179d511.yaml
Comment (mfpierre) [inline releasenotes/notes/AddPodCheck-355c270d2179d511.yaml:11]: ditto for the core check, I'd phrase it as "The orchestrator check is moving from the Process Agent to the Core Agent" or something around those line instead

File: pkg/collector/corechecks/orchestrator/pod/pod.go
Comment (vickenty) [inline pkg/collector/corechecks/orchestrator/pod/pod.go:84]: This still produces an un-actionable warning for every installation.

Could you please use [`staticconfig`](https://github.com/datadog/datadog-agent/blob/4fe56c9b19365fe294f1683350e831a2f16b404c/pkg/autodiscovery/listeners/staticconfig.go) to conditionally enable the check if the feautre is enabled in the agent?

**Review by vickenty — CHANGES_REQUESTED** [review]

File: pkg/process/checks/pod.go
Comment (JLineaweaver) [reply pkg/process/checks/pod.go]: [78fa929](https://github.com/DataDog/datadog-agent/pull/19478/commits/78fa92963f08ddbb3e6049590de1b8a0cf0a910e)

File: releasenotes/notes/AddPodCheck-355c270d2179d511.yaml
Comment (JLineaweaver) [reply releasenotes/notes/AddPodCheck-355c270d2179d511.yaml]: [78fa929](https://github.com/DataDog/datadog-agent/pull/19478/commits/78fa92963f08ddbb3e6049590de1b8a0cf0a910e)

File: releasenotes/notes/AddPodCheck-355c270d2179d511.yaml
Comment (vickenty) [inline releasenotes/notes/AddPodCheck-355c270d2179d511.yaml:11]: Changelog uses RST instead of Markdown:
```suggestion
    The orchestrator check is moving from the Process Agent to the Core Agent. In the next release, this new check will replace the current pod check in the Process Agent. You can start using this new check now by manually setting the environment variable ``DD_ORCHESTRATOR_EXPLORER_RUN_ON_NODE_AGENT`` to ``true``.
```

**Review by vickenty — APPROVED** [review]

**Review by mfpierre — APPROVED** [review]

**Review by mfpierre — APPROVED** [review]

**Review by fisherevans — APPROVED** [review]

**Review by sgnn7 — APPROVED** [review]

**Pull request merged**
